crypto/rsa/rsa_lib.c

```diff
@@ -1244,27 +1056,20 @@ int EVP_PKEY_CTX_set0_rsa_oaep_label(EVP_PKEY_CTX *ctx, void *label, int llen)
     }
 
     /* If key type not RSA return error */
-    if (ctx->pmeth != NULL && ctx->pmeth->pkey_id != EVP_PKEY_RSA)
+    if (!EVP_PKEY_CTX_is_a(ctx, "RSA"))
         return -1;
 
-    /* TODO(3.0): Remove this eventually when no more legacy */
-    if (ctx->op.ciph.ciphprovctx == NULL)
-        return EVP_PKEY_CTX_ctrl(ctx, EVP_PKEY_RSA, EVP_PKEY_OP_TYPE_CRYPT,
-                                 EVP_PKEY_CTRL_RSA_OAEP_LABEL, llen,
-                                 (void *)label);
+    /* TODO(3.0) Shouldn't a set0 translate into setting an OCTET_PTR? */
```

Comment:
IMO no. As I'm contemplating in #12985

Reply:
Okie.  I think I'll leave that comment all the same.  We should take up the question in the next OTC meeting, as it's a fairly generic one.  However, that shouldn't block this PR, can we agree on that?